data/transactions/logic/assembler.go

```diff
@@ -1531,6 +1534,38 @@ func (ops *OpStream) trackStack(args StackTypes, returns StackTypes, instruction
 	}
 }
 
+func processFields(ops *OpStream, fields []string) (current, next []string) {
+	for i := 0; i < len(fields); i++ {
+		field := fields[i]
+		replacement, ok := ops.macros[field]
+		if ok {
+			fields = append(fields[0:i], append(replacement, fields[i+1:]...)...)
+			i--
+			continue
+		}
+		if string(field[len(field)-1]) == ";" {
+			field = field[0 : len(field)-1]
+			replacement, ok = ops.macros[field]
+			if ok {
+				last := replacement[len(replacement)-1]
+				fields = append(fields[0:i], append(replacement, fields[i+1:]...)...)
+				fields[i+len(replacement)-1] = last + ";"
+				i--
+				continue
+			}
```

Comment:
Seems like if you look for and strip off the semi-colon first, you can probbaly avoid doing the expansion work twice.